### ISSUE TITLE
Vulkan: Godot cannot find "VKLayer_nsight-sys_*" even though it exists.

### ISSUE THREAD

**Issue by totlmstr** [godotengine/godot#36688]
**Godot version:**
4.0 master ([tested with recent commit](https://github.com/godotengine/godot/commit/2a15304572516a8df2dabd9561c959832badf60f) as `release_debug`)

**OS/device including version:**
Windows 10 x86_64
NVIDIA Geforce 1070 (442.50)

**Issue description:**
Godot is reporting that it cannot find (for my system) `VkLayer_nsight-sys_windows.json`. The specific error message:

```
Godot Engine v4.0.dev.custom_build.2a1530457 - https://godotengine.org
ERROR: ERROR : GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
        loader_get_json: Failed to open JSON file C:\Program Files\NVIDIA Corporation\Nsight Systems 2019.5.2\Target-Windows\x86_64\VkLayers\VkLayer_nsight-sys_windows.json
        Objects - 1
                Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 2278264767408
   at: VulkanContext::_debug_messenger_callback (drivers\vulkan\vulkan_context.cpp:142)
```

In Nsight Systems 2019.5.2 and 2020.1.1 (latest), the actual (and default) location where it is found is in (using 2020.1.1 and Windows x86_64):

`%PROGRAMFILES%/NVIDIA Corportation/Nsight Systems.2020.1.1/target-windows-x64/vulkan-layers/VkLayer_nsight-sys_windows.json`

Both Nsight Systems I have installed are similarly structured that way.

**Steps to reproduce:**
Simply load up Godot.

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
Godot simply relies on Vulkan-Loader to find ICDs on the user system. If it gets them wrong here, I suspect you have some registry keys that are out of sync with your current Nsight setup. Other Vulkan applications might fail similarly.

**Comment by totlmstr** [comment]
Had to use [RegScanner](https://www.nirsoft.net/utils/regscanner.html) to find the culprit, and it turns out that it was a registry key that was wrong. The specific location is:

`HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\Vulkan\ExplicitLayers`

That above-mentioned wrong location is stored in this folder. It contains Vulkan SDK JSON locations, the 2020 version, and the 2019 version. The location in the 2020 version was correct, so I uninstalled the 2019 version. For some reason, the 2019 key did not delete itself when I uninstalled the 2019 version, so I had to manually delete it in Registry Editor. Now, there are no errors from that location.
